common/tools/hextobin/build.sh

```diff
@@ -11,7 +11,7 @@ THIS_SCRIPT="$(readlink -f "${BASH_SOURCE[0]}")"
 
 builder_describe "Build hextobin" clean configure build
 builder_describe_outputs \
-  configure /node_modules \
+  configure /common/tools/hextobin/node_modules/commander \
```

Comment:
I should say "half-clean" then. :smile: I often run a git clean on the subdirectory I'm working on in order to save some build time. 

Yes, I realize that this is possibly fragile if we move files around and we can adjust then. Just checking for the existence of `/node_modules` is too broad IMO.

Reply:
> Yes, I realize that this is possibly fragile if we move files around and we can adjust then.

The problem is that node_modules files can move around without us being very aware of it happening, because the choice of location is quite dependent on npm and whether or not we've hoisted packages (and if packages are shared across projects). It is very likely we'd simply miss the change and then we'd end up having the configure step running on every build until we notice the build slowdown.

> Just checking for the existence of `/node_modules` is too broad IMO.

I guess so -- but it's what we are doing for all the other node/web projects, so this is a broader issue across many projects. We could consider having a separate '.configured' flag file created on each project as a more robust alternative that doesn't depend on vagaries of npm (and then we need to touch that file in build.sh of course):

```suggestion
  configure /common/tools/hextobin/.configured \
```

\</bikeshed>

Reply:
Good discussion! See #12481 for a solution that picks up your idea.